src/grants/client_credentials.grant.ts

```diff
@@ -30,12 +30,44 @@ export class ClientCredentialsGrant extends AbstractGrant {
   }
 
   async respondToIntrospectRequest(req: RequestInterface): Promise<ResponseInterface> {
-    // introspection is authenticated via client_credentials, but we don't want to require this param in the request
-    req.body["grant_type"] = "client_credentials";
+    const { parsedToken, oauthToken, expiresAt, tokenType } = await this.tokenFromRequest(req);
+
+    const active = expiresAt > new Date();
+
+    const responseBody: OAuthTokenIntrospectionResponse = active
+      ? {
+          active: true,
+          scope: oauthToken.scopes.map(s => s.name).join(this.options.scopeDelimiter),
+          client_id: oauthToken.client.id,
+          token_type: tokenType,
+          ...parsedToken,
+        }
+      : { active: false };
+
+    const response = new OAuthResponse();
+    response.body = responseBody;
+    return response;
+  }
+
+  canRespondToRevokeRequest(request: RequestInterface): boolean {
```

Comment:
the server will assume that the token is an access_token, so all other revoke requests besides auth_code should come to the client credentials grant and if it is missing a token, it will throw, and then if it is not an access/refresh token, it will throw, etc.